src/components/header/header.tsx

```diff
@@ -16,6 +17,16 @@ export interface HeaderProps {
 export const Header = ({ className }: HeaderProps) => {
     const cart = useCartData();
     const cartOpener = useCartOpen();
+    const navigate = useNavigate();
+
+    const onSubmit: React.FormEventHandler<HTMLFormElement> = (event) => {
```

Comment:
It would be nicer if instead of the raw event we received the search term that is already validated. Keeps the Header component cleaner, and avoids repetition if the `SearchInput` is used in multiple places.

```ts
const onSearchSubmit = (search: string) => {
    navigate(`/products/all-products?search=${encodeURIComponent(search)}`);
};

<SearchInput onSubmit={onSearchSubmit} />
```

Reply:
done. much better